### ISSUE TITLE
move positioning code from MaterialMenu.prototype.handleForClick_ into public api

### ISSUE THREAD

**Issue by rj33** [google/material-design-lite#1701]
MaterialMenu exposes a show() command in its public API, but it would be a lot easier to use if the positioning code that is currently in the handleForClick_ private handler of the click event was available publicly, as show() does no positioning, so you are left to essentially duplicate the positioning from inside handleForClick_ in your own code. 

It might be simplest to put the positioning into show() as the positioning code does check for the case where the user is handling the position themselves due to not assigning a position class, but the user might want to supply a position class , so it is perhaps best to add a separate setPosition() call.

I wasn't sure, but it also looks like handleForClick_ might be needlessly doing positioning when the menu is only going to be hidden (i.e. the positioning code is all done before the toggle call is made, so is down irrespectively of whether you are toggling the menu on or off).

**Comment by surma** [comment]
@rj33 Very valid point! I’ll see if we can incorporate that without breaking anything, but your suggestion sounds good to me.

**Comment by rj33** [comment]
Thanks @surma. A quick side question, is hover/mouseover style menus something that would be considered, or is that kind of thing seen as undesirable in terms of MD specs? I didn't see any "dont's" in the spec, but I couldn't find a mention of that type of behaviour, or many examples in live implementations. (I noticed the show() behaviour described here when attempting to implement my own 'expand menu on mouseover' externally).

**Comment by surma** [comment]
Sorry, I can’t follow. We have hover styles
![screenshot 2015-10-07 17 14 59](https://cloud.githubusercontent.com/assets/234957/10343695/f654220a-6d16-11e5-9ff3-01301aa70ea1.png)

Generally, though, hover is not as present in the MD spec as hover is not a thing on mobile.

**Comment by surma** [comment]
No, triggering actions via hover is definitely against the MD spec, as hover is not a thing on mobile (tooltips are a corner-case already).

**Comment by rj33** [comment]
Thanks, that makes sense.
